### DISCUSSION TITLE
Handing off buffers in CURLOPT_READFUNCTION

### DISCUSSION THREAD

**Discussion by laramiel** [curl/curl#15554] [General]
I'm trying to reduce the number of copies which happen in my network code.
Is there a way for libcurl to hand off the buffers, maybe along with a deallocation function, in e.g. CURLOPT_READFUNCTION or similar?

**Comment by dfandrich** [comment]
The idea of adding a zero-copy interface to libcurl is years old, but nobody
has felt it worthwhile to add yet. With the prevalence of encryption these days
making true zero-copy almost impossible any more, along with the increase in memory
bandwidth, the improvement likely wouldn't be as great today as it likely would
have been in years past. But, if someone were to do it, I'm guessing it would
have a positive reception.
